libdnf/repo/Repo.cpp

```diff
@@ -703,8 +780,8 @@ static std::vector<Key> rawkey2infos(int fd) {
     std::unique_ptr<std::remove_pointer<gpgme_ctx_t>::type> context(ctx);
 
     // set GPG home dir
-    char tmpdir[] = "/tmp/tmpdir.XXXXXX";
-    mkdtemp(tmpdir);
+    char tmpdir[] = "/tmp/libdnf.XXXXXX";
+    create_temporary_directory(tmpdir);
```

Comment:
Although not directly connected with this PR, while reviewing the [RHEL 9 version](https://github.com/rpm-software-management/libdnf/pull/1634) of these changes, I noticed that failures in the `create_temporary_directory` calls consistently result in throwing a `RepoError` there. In contrast, in the RHEL 8 case, these errors are sometimes ignored, like here, leading to different behavior in both systems. Not sure if it's intentional, but it appears that [this](https://github.com/rpm-software-management/libdnf/pull/1283) related PR for the RHEL 8 branch is missing.

@kontura Do you have an idea why is that? 

But anyway, this PR is not changing this behavior, so I am in favor or merging.

Reply:
If I understand correctly you are asking why the PR is missing. I think it is simply because we didn't do a rebase in RHEL 8 since it was merged and there was no request for it to be backported.
There is this [bug](https://bugzilla.redhat.com/show_bug.cgi?id=1938761) for it but that is for RHEL 9.